be/src/storage/tablet_updates.cpp

```diff
@@ -252,6 +252,7 @@ Status TabletUpdates::_load_from_pb(const TabletUpdatesPB& tablet_updates_pb) {
 
     RETURN_IF_ERROR(_load_meta_and_log(tablet_updates_pb));
 
+    _rowset_stats.clear();
```

Comment:
What will be happened if it is not cleared?

Reply:
> What will be happened if it is not cleared?

When we do alter job for primary key table, we will reallocate rowset id for each rowset and the rowset id maybe conflict with the old rowsets. And we will delete the old rowsets and reload the new rowsets. So if we don't clear `_rowset_stats`, it maybe keep the stats of old rowset which will cause the following check failed.  This can lead to errors of judgment in some normal scenarios.